opentelemetry-sdk/src/metrics/periodic_reader.rs

```diff
@@ -112,93 +105,194 @@ where
 
     /// Create a [PeriodicReader] with the given config.
     pub fn build(self) -> PeriodicReader {
-        let (message_sender, message_receiver) = mpsc::channel(256);
-
-        let worker = move |reader: &PeriodicReader| {
-            let runtime = self.runtime.clone();
-            let reader = reader.clone();
-            self.runtime.spawn(Box::pin(async move {
-                let ticker = runtime
-                    .interval(self.interval)
-                    .skip(1) // The ticker is fired immediately, so we should skip the first one to align with the interval.
-                    .map(|_| Message::Export);
-                let messages = Box::pin(stream::select(message_receiver, ticker));
-                PeriodicReaderWorker {
-                    reader,
-                    timeout: self.timeout,
-                    runtime,
-                    rm: ResourceMetrics {
-                        resource: Resource::empty(),
-                        scope_metrics: Vec::new(),
-                    },
-                }
-                .run(messages)
-                .await
-            }));
-        };
-
-        PeriodicReader {
-            exporter: Arc::new(self.exporter),
-            inner: Arc::new(Mutex::new(PeriodicReaderInner {
-                message_sender,
-                is_shutdown: false,
-                sdk_producer_or_worker: ProducerOrWorker::Worker(Box::new(worker)),
-            })),
-        }
+        PeriodicReader::new(self.exporter, self.interval, self.timeout)
     }
 }
 
 /// A [MetricReader] that continuously collects and exports metric data at a set
 /// interval.
 ///
-/// By default it will collect and export data every 60 seconds, and will cancel
-/// export attempts that exceed 30 seconds. The export time is not counted
-/// towards the interval between attempts.
+/// By default, PeriodicReader will collect and export data every
+/// 60 seconds. The export time is not counted towards the interval between
```

Comment:
"The export time is not counted towards the interval between attempts", does this mean if the export is too slow like exeeding 60 seconds, multiple export then can to be outstanding(pending), and new export will still issued after another 60 seconds?

Reply:
I have left a code comment to explain the behavior when export itself takes more than the interval. This PR's behavior is to trigger the next one immediately and then continue to honor the internal for the next one onwards.

Reply:
@ThomsonTan any additional comments on this?